runtime_constraints/README.md

```diff
@@ -0,0 +1,11 @@
+# Cluster Runtime Constraints
```

Comment:
Does this belong in `docs.go`? i.e. looks like something we want in the go docs.

Reply:
Could you please be more specific for me: do you mean there's a docs place in this repo?  do you mean upstream? do you mean docstrings in the code?

My intent here is to add developer documentation - just so people get the context of why that file exists there. Once we have the _right_ way of doing things, then all of this should go away.

It is still my intention to add user facing docs though - but I was thinking of doing that in the upstream

Reply:
Sorry -- Go doc strings; e.g. https://github.com/kubernetes-sigs/controller-runtime/blob/master/doc.go autogenerates developer docs at https://pkg.go.dev/sigs.k8s.io/controller-runtime